core/src/main/java/feast/core/model/Entity.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2019 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.model;
+
+import feast.core.FeatureSetProto.EntitySpec;
+import feast.types.ValueProto.ValueType;
+import java.util.Objects;
+import lombok.Getter;
+import lombok.Setter;
+
+/** Feast entity object. Contains name, type as well as domain metadata about the entity. */
+@Getter
+@Setter
+@javax.persistence.Entity
+public class Entity extends Field {
+
+  public Entity() {}
+
+  public Entity(FieldId fieldId) {
```

Comment:
We should possibly make these static factory methods like `fromProto`. It would make the API a bit cleaner.